p2p/netaddress.go

```diff
@@ -232,17 +235,37 @@ func (na *NetAddress) DialString() string {
 }
 
 // Dial calls net.Dial on the address.
-func (na *NetAddress) Dial() (net.Conn, error) {
-	conn, err := net.Dial("tcp", na.DialString())
+// TODO: add TLS stuff.
+// Note: this one is not used in the code, the DialTimeout is used instead.
+func (na *NetAddress) Dial(ctx context.Context) (quic.Connection, error) {
+	tlsConfig := tls.Config{
+		MinVersion:         tls.VersionTLS13,
+		InsecureSkipVerify: true,
```

Comment:
IIUC, this means TLS is disabled.

Reply:
this function is not used anywhere, that's why I didn't fix its implementation. Maybe later